gems/smithy-client/lib/smithy-client/plugins/stub_responses.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    module Plugins
+      # @api private
+      class StubResponses < Plugin
+        option(
+          :stub_responses,
+          default: false,
+          doc_type: 'Boolean',
+          docstring: <<~DOCS)
+            When true, the client will return stubbed responses instead of networking requests.
+            By default fake responses are generated and returned. You can specify the response data
+            to return or errors to raise by calling {Stubs#stub_responses}.
+            @see Stubs
+          DOCS
+
+        option(:stubs) { {} }
+        option(:stubs_mutex) { Mutex.new }
+        option(:api_requests) { [] }
+        option(:api_requests_mutex) { Mutex.new }
+
+        def add_handlers(handlers, config)
```

Comment:
This is just for curiosity- do you prefer to have `add_handlers` on the top or bottom? Or does it really depend on context?

Reply:
I just follow the order that was defined in the Plugin class.